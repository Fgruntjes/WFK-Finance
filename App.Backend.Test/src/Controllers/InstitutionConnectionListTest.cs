using App.Lib.Data.Entity;

namespace App.Backend.Test.Controllers;

public class InstitutionConnectionListTest : IClassFixture<InstitutionConnectionListFixture>
{
    private readonly InstitutionConnectionListFixture _fixture;

    public InstitutionConnectionListTest(InstitutionConnectionListFixture fixture)
    {
        _fixture = fixture;
    }

    [Fact]
    public async Task WithoutSkipLimit()
    {
        // Act
        var result = await _fixture.Client.ExecuteQuery();

        // Assert
        result.MatchSnapshot();
    }

    [Fact]
    public async Task WithSkipLimit()
    {
        // Act
        var result = await _fixture.Client.ExecuteQuery(new { offset = 1, limit = 1 });

        // Assert
        result.MatchSnapshot();
    }

    [Fact]
    public async Task OnlyWithinOrganisation()
    {
        // Arrange
        var connectionEntity = new InstitutionConnectionEntity()
        {
            ExternalId = $"SomeExternalId-organisation-missmatch-0",
            ConnectUrl = new Uri($"https://www.example-organisation-missmatch-0.com/"),
            InstitutionId = _fixture.InstitutionEntity.Id,
            OrganisationId = _fixture.AltOrganisationId,
        };
        _fixture.Database.SeedData(c =>
        {
            c.InstitutionConnections.Add(connectionEntity);
        });

        // Act
        var result = await _fixture.Client.ExecuteQuery();

        // Assert
        result.MatchSnapshot();
    }
}