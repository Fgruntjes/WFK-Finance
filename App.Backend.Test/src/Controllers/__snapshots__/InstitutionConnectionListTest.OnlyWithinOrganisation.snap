{
  "data": {
    "institutionConnection": {
      "list": {
        "items": [
          {
            "externalId": "SomeExternalId-organisation-match-0"
          },
          {
            "externalId": "SomeExternalId-organisation-list-0"
          },
          {
            "externalId": "SomeExternalId-organisation-list-1"
          },
          {
            "externalId": "SomeExternalId-organisation-list-2"
          },
          {
            "externalId": "SomeExternalId-organisation-list-3"
          },
          {
            "externalId": "SomeExternalId-organisation-list-4"
          },
          {
            "externalId": "SomeExternalId-organisation-list-5"
          },
          {
            "externalId": "SomeExternalId-organisation-list-6"
          },
          {
            "externalId": "SomeExternalId-organisation-list-7"
          },
          {
            "externalId": "SomeExternalId-organisation-list-8"
          },
          {
            "externalId": "SomeExternalId-organisation-list-9"
          },
          {
            "externalId": "SomeExternalId-organisation-list-10"
          },
          {
            "externalId": "SomeExternalId-organisation-list-11"
          },
          {
            "externalId": "SomeExternalId-organisation-list-12"
          },
          {
            "externalId": "SomeExternalId-organisation-list-13"
          },
          {
            "externalId": "SomeExternalId-organisation-list-14"
          },
          {
            "externalId": "SomeExternalId-organisation-list-15"
          },
          {
            "externalId": "SomeExternalId-organisation-list-16"
          },
          {
            "externalId": "SomeExternalId-organisation-list-17"
          },
          {
            "externalId": "SomeExternalId-organisation-list-18"
          },
          {
            "externalId": "SomeExternalId-organisation-list-19"
          },
          {
            "externalId": "SomeExternalId-organisation-list-20"
          },
          {
            "externalId": "SomeExternalId-organisation-list-21"
          },
          {
            "externalId": "SomeExternalId-organisation-list-22"
          },
          {
            "externalId": "SomeExternalId-organisation-list-23"
          }
        ],
        "totalCount": 31
      }
    }
  }
}
